Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveHM_2nd.cpp

```diff
@@ -651,32 +651,62 @@ void FiniteDifferenceSolver::MacroscopicEvolveHMCartesian_2nd(
             Box const &tbx = mfi.tilebox(Hfield[0]->ixType().toIntVect());
             Box const &tby = mfi.tilebox(Hfield[1]->ixType().toIntVect());
             Box const &tbz = mfi.tilebox(Hfield[2]->ixType().toIntVect());
+            
+            // read in Ms to decide if the grid is magnetic or not
+            auto& mag_Ms_mf = macroscopic_properties->getmag_Ms_mf();
+            Array4<Real> const& mag_Ms_arr = mag_Ms_mf.array(mfi);
+            //if (mag_Ms_mf.min(1,0) == 0){
+                auto& mu_mf = macroscopic_properties->getmu_mf();
+                Array4<Real> const& mu_arr = mu_mf.array(mfi);
+            //}
 
             amrex::Real const mu0_inv = 1. / PhysConst::mu0;
+
             // Loop over the cells and update the fields
             amrex::ParallelFor(tbx, tby, tbz,
 
                 [=] AMREX_GPU_DEVICE(int i, int j, int k) {
-                    Hx(i, j, k) = Hx_old(i, j, k) + mu0_inv * dt * T_Algo::UpwardDz(Ey, coefs_z, n_coefs_z, i, j, k)
-                                                  - mu0_inv * dt * T_Algo::UpwardDy(Ez, coefs_y, n_coefs_y, i, j, k);
-                    if (coupling == 1) {
-                        Hx(i, j, k) += - M_xface(i, j, k, 0) + M_xface_old(i, j, k, 0);
+                    Real mag_Ms_arrx = MacroscopicProperties::macro_avg_to_face(i, j, k, amrex::IntVect(1,0,0), mag_Ms_arr);
+                    if (mag_Ms_arrx == 0){ // nonmagnetic region
```

Comment:
It's good practice in C++ to use "0." instead of "0" when comparing to a real number (not integer)